tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/RuntimeBrokerTests.cs

```diff
@@ -239,6 +241,36 @@ public async Task WamListWindowsWorkAndSchoolAccountsAsync()
             var account = accounts.FirstOrDefault();
             Assert.AreEqual(labResponse.User.Upn, account.Username);
         }
+
+        [DataTestMethod]
+        [DataRow("")]
+        [DataRow(" ")]
+        [DataRow(null)]
+        [RunOn(TargetFrameworks.NetStandard | TargetFrameworks.NetCore)]
+        public async Task WamNoScopesAsync(string scopes)
+        {
+            string expectedErrorMessage = "Scopes are a required authentication parameter";
+            var labResponse = await LabUserHelper.GetDefaultUserAsync().ConfigureAwait(false);
+            
+            IntPtr intPtr = GetForegroundWindow();
+
+            Func<IntPtr> windowHandleProvider = () => intPtr;
+
+            IPublicClientApplication pca = PublicClientApplicationBuilder
+               .Create("04f0c124-f2bc-4f59-8241-bf6df9866bbd")
+               .WithAuthority("https://login.microsoftonline.com/organizations")
+               .WithParentActivityOrWindow(windowHandleProvider)
+               .WithBrokerPreview()
+               .Build();
+
+            var ex = await AssertException.TaskThrowsAsync<MsalServiceException>(
```

Comment:
I was under the impression that MSAL C++ now allows no scopes?

Reply:
Per discussion I had with @MSamWils and @logdog16, scopes is a required parameter for MSAL C++. But the magic scopes can be passed and token retrieved. There was a bug with magic scopes and signout but that is now fixed. @MSamWils can you please confirm the behavior

Reply:
That's correct, passing the magic scope should works fine including sign out and sign back in.

Reply:
I thought if the user passes empty scopes, we add [default/reserved](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/80c7824871bd6369e740479ba0497850bfbe51de/src/client/Microsoft.Identity.Client/OAuth2/OAuthConstants.cs#L101) scopes, like [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/80c7824871bd6369e740479ba0497850bfbe51de/src/client/Microsoft.Identity.Client/OAuth2/TokenClient.cs#L62). Seems like for old WAM, we check for null only, not empty, see [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/80c7824871bd6369e740479ba0497850bfbe51de/src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/AadPlugin.cs#L266).

So for MSAL runtime, we should pass default scopes, if user scopes are null/empty. Yes?

Reply:
I thought if the user passes empty scopes, we add [default/reserved](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/80c7824871bd6369e740479ba0497850bfbe51de/src/client/Microsoft.Identity.Client/OAuth2/OAuthConstants.cs#L101) scopes, like [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/80c7824871bd6369e740479ba0497850bfbe51de/src/client/Microsoft.Identity.Client/OAuth2/TokenClient.cs#L62). Seems like for old WAM, we check for null only, not empty, see [here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/80c7824871bd6369e740479ba0497850bfbe51de/src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/AadPlugin.cs#L266).

So for MSAL runtime, we should pass default scopes, if user scopes are null/empty. Yes?

Reply:
sounds good, let me fix it

Reply:
@pmaytak please review